tasties_app/models.py

```diff
@@ -1,3 +1,90 @@
-# from django.db import models
+from django.db import models
+from django.utils import timezone
+from django.contrib.auth.models import User
+from django.core.validators import MinValueValidator, MinLengthValidator, MaxValueValidator
 
-# Create your models here.
+
+class Category(models.Model):
+    category_name = models.CharField(max_length=16, unique=True)
+
+
+class Recipe(models.Model):
+    title = models.CharField(max_length=64, validators=[MinLengthValidator(1)], unique=True)
+    author_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    categories = models.ManyToManyField(Category)
+    description = models.CharField(max_length=512, validators=[MinLengthValidator(1)])
+    directions = models.CharField(max_length=65536, validators=[MinLengthValidator(1)])
+    publication_date = models.DateField(default=timezone.now)
+    minutes_to_make = models.IntegerField(validators=[MinValueValidator(1)])
+    recipe_picture = models.ImageField(upload_to='recipe_pictures')
+
+
+class Ingredient(models.Model):
+    """
+    Ingredient Model - represents an ingredient in a recipe.
+    Fields:
+        UNIT_CHOICES - a list of valid measurement units
+        recipe_id - the recipe that the ingredient belongs to
+        amount - the amount of the ingredient
+        measurement_unit - the unit of measurement for the ingredient
+        description - a description of the ingredient
+    """
+
+    WHOLE = "WHOLE"
+    FLOZ = "FLOZ"
+    TSP = "TSP"
+    OZ = "OZ"
+    CUP = "CUP"
+    GRAM = "GRAM"
+    ML = "ML"
+    KG = "KG"
+    L = "L"
+    U = "U"
+    UNIT_CHOICES = ((WHOLE, 'Whole'),
+                    (FLOZ, 'Fluid Ounce'),
+                    (TSP, 'Teaspoon'),
+                    (OZ, 'Ounce'),
+                    (CUP, 'Cup'),
+                    (GRAM, 'Gram'),
+                    (ML, 'Milliliter'),
+                    (KG, 'Kilogram'),
+                    (L, 'Liter'),
+                    (U, 'Units'),
```

Comment:
I see you've added the additional measurement units back. This will require us syncing each other's models.py and migrations to accommodate. Regardless, you've added a "Units" choice - my intention in including "Whole" (which you didn't remove) was to accommodate keeping track of ingredients like "3 Chicken breasts", "6 Whole Apples", "1 large Pumpkin" etc.; Does "Unit" do something other than that, or provide more functionality?